crates/forge/tests/data/file_reading/test.cairo

```diff
@@ -48,35 +48,78 @@ struct D {
     e: u8
 }
 
+#[derive(Serde, Drop, PartialEq)]
+struct E {
+    a: felt252,
+    b: F
+}
+#[derive(Serde, Drop, PartialEq)]
+struct F {
+    c: felt252,
+    d: u8,
+    e: G
+}
+#[derive(Serde, Drop, PartialEq)]
+struct G {
+    c: felt252,
+}
+
 
 #[test]
 fn valid_content_and_same_content_no_matter_whitespaces() {
     let file = FileTrait::new('data/valid.txt');
-    let content = TxtParser::<A>::deserialize_txt(@file).unwrap();
+    let content = Parser::<A>::deserialize_txt(@file).unwrap();
     let expected = A {
         a: 1, nested_b: B {
             nested_c: C { c: u256 { low: 'hello', high: 3 } }
             }, nested_d: D {
             d: 'world', e: 0
         }, f: 3618502788666131213697322783095070105623107215331596699973092056135872020480,
     };
+    assert(content.f == expected.f, '')
 }
 
 #[test]
 fn serialization() {
     let file = FileTrait::new('data/valid.txt');
-    let content = parse_txt(@file);
+    let content = read_txt(@file);
     compare_with_expected_content(content);
 
     let file = FileTrait::new('data/valid_diff_spaces.txt');
-    let content = parse_txt(@file);
+    let content = read_txt(@file);
+    compare_with_expected_content(content);
+}
+
+#[test]
+fn json_serialization() {
+    let file = FileTrait::new('data/json/valid.json');
+    let content = read_json(@file);
     compare_with_expected_content(content);
 }
 
+#[test]
+fn invalid_json() {
+    let file = FileTrait::new('data/json/invalid.json');
+    let content = read_json(@file);
+    assert(1 == 1, '');
+}
+
+#[test]
+fn json_deserialization() {
+    let file = FileTrait::new('data/json/nested_valid.json');
+    let content = Parser::<E>::deserialize_json(@file).unwrap();
+
+    let mut output_array = ArrayTrait::new();
+    let serialized = content.serialize(ref output_array);
+    assert(content.a == 23, '');
+    assert(content.b.c == 'test', '');
+    assert(content.b.e.c == 2, '');
```

Comment:
Shouldn't 2 be equal to `b.e.a`? Or are the fields of structs defined in intentionally different than these in `nested_valid.json`?

Reply:
yep I did it intentionally because the names of JSON keys has only impact on values order